packages/x-data-grid/src/tests/quickFiltering.DataGrid.test.tsx

```diff
@@ -165,188 +168,162 @@ describe('<DataGrid /> - Quick filter', () => {
+        'false',
+      );
     });
 
     it('should be expanded by default if there is a value', () => {
       render(<TestCase filterModel={{ items: [], quickFilterValues: ['adidas'] }} />);
 
       expect(screen.getByRole<HTMLInputElement>('searchbox').value).to.equal('adidas');
       expect(screen.getByRole<HTMLInputElement>('searchbox').tabIndex).to.equal(0);
-      expect(
-        screen
-          .getByRole<HTMLButtonElement>('button', { name: 'Search' })
-          .getAttribute('aria-expanded'),
-      ).to.equal('true');
+      expect(screen.getByRole('button', { name: 'Search' }).getAttribute('aria-expanded')).to.equal(
+        'true',
+      );
     });
 
-    it('should expand when the trigger is clicked', () => {
-      render(<TestCase />);
+    it('should expand when the trigger is clicked', async () => {
+      const { user } = render(<TestCase />);
 
-      fireEvent.click(screen.getByRole<HTMLButtonElement>('button', { name: 'Search' }));
+      await user.click(screen.getByRole('button', { name: 'Search' }));
 
-      expect(
-        screen
-          .getByRole<HTMLButtonElement>('button', { name: 'Search' })
-          .getAttribute('aria-expanded'),
-      ).to.equal('true');
+      expect(screen.getByRole('button', { name: 'Search' }).getAttribute('aria-expanded')).to.equal(
+        'true',
+      );
     });
 
-    it('should expand when the input changes value', () => {
-      render(<TestCase />);
-
-      fireEvent.focus(screen.getByRole<HTMLInputElement>('searchbox'));
+    it('should expand when the input changes value', async () => {
+      const { user } = render(<TestCase />);
 
-      fireEvent.change(screen.getByRole<HTMLInputElement>('searchbox'), {
-        target: { value: 'adidas' },
-      });
+      await user.type(screen.getByRole<HTMLInputElement>('searchbox'), 'adidas');
 
-      expect(
-        screen
-          .getByRole<HTMLButtonElement>('button', { name: 'Search' })
-          .getAttribute('aria-expanded'),
-      ).to.equal('true');
+      expect(screen.getByRole('button', { name: 'Search' }).getAttribute('aria-expanded')).to.equal(
+        'true',
+      );
     });
 
-    it('should collapse when the input is blurred with no value', () => {
-      render(<TestCase />);
+    describe('with fake timers', () => {
+      // TODO: temporary for vitest. Can move to `vi.useFakeTimers`
+      let timer: SinonFakeTimers | null = null;
+
+      beforeEach(() => {
+        timer = useFakeTimers({ shouldClearNativeTimers: true });
+      });
 
-      fireEvent.click(screen.getByRole<HTMLButtonElement>('button', { name: 'Search' }));
+      afterEach(() => {
+        timer?.restore();
+      });
 
-      expect(
-        screen
-          .getByRole<HTMLButtonElement>('button', { name: 'Search' })
-          .getAttribute('aria-expanded'),
-      ).to.equal('true');
+      it('should collapse when the escape key is pressed with no value', () => {
```

Comment:
This test doesn't work well with userEvents on mocha, even though it works in vitest. I just wrapped it in the fake timers for now.